lib/route.js

```diff
@@ -173,11 +173,23 @@ function buildRouting (options) {
   }
 
   function findRoute (options) {
-    return router.find(
+    const route = router.find(
       options.method,
       options.url || '',
       options.constraints
     )
+    if (route) {
+      // we must reduce the expose surface, otherwise
+      // we provide the ability for the user to modify
+      // all the route and server information in runtime
+      return {
```

Comment:
Should we deepFreeze the Object?

Reply:
I generally avoid freezing because it wrecks performance.

Reply:
> Should we deepFreeze the Object?

It doesn't matter on this case, `params` and `searchParams` is newly created object based on the current arguments.
`handler` is a function. There is no way to affect the core by this given values.